src/trio/_socket.py

```diff
@@ -169,7 +169,8 @@ def set_custom_socket_factory(
 # getaddrinfo and friends
 ################################################################
 
-_NUMERIC_ONLY = _stdlib_socket.AI_NUMERICHOST | _stdlib_socket.AI_NUMERICSERV
+_NUMERIC_ONLY = _stdlib_socket.AI_NUMERICHOST
+_NUMERIC_ONLY |= getattr(_stdlib_socket, "AI_NUMERICSERV", 0)
```

Comment:
perhaps good to add a comment explaining why this is needed / when the workaround can be removed

Reply:
@CoolCat467 It does not exist on any macOS on any arch before 10.6 release. In practice, this will mostly be 10.5 and 10.4, on ppc and i386.